dropshot/tests/test_streaming.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 Oxide Computer Company
+
+//! Test cases for streaming reqeusts.
+
+use dropshot::{endpoint, ApiDescription, HttpError, RequestContext};
+use http::{Method, Response, StatusCode};
+use hyper::{body::HttpBody, Body};
+use hyper_staticfile::FileBytesStream;
+use std::sync::Arc;
+use tokio::io::{AsyncSeekExt, AsyncWriteExt};
+
+#[macro_use]
+extern crate slog;
+
+mod common;
+
+fn api() -> ApiDescription<usize> {
+    let mut api = ApiDescription::new();
+    api.register(api_streaming).unwrap();
+    api
+}
+
+const BUF_SIZE: usize = 8192;
+const BUF_COUNT: usize = 128;
+
+#[endpoint {
+    method = GET,
+    path = "/streaming",
+}]
+async fn api_streaming(
+    _rqctx: Arc<RequestContext<usize>>,
+) -> Result<Response<Body>, HttpError> {
+    let mut file = tempfile::tempfile()
```

Comment:
Is there any particular reason to use an actual file instead of, say, a [Cursor](https://doc.rust-lang.org/std/io/struct.Cursor.html)?  Or maybe better: a custom Stream that just emits a fixed number of arbitrary bytes?

It'd be neat to have this stream emit some number of bytes that's likely to be larger than available memory.  If the test works, that would validate that we're not at any point buffering the whole contents of the response anywhere.

Reply:
I used a file to show integration with the [hyper_staticfile::FileBytesStream](https://docs.rs/hyper-staticfile/0.8.0/hyper_staticfile/struct.FileBytesStream.html), since that was a use case that motivated adding this support - we have a file server example in here, but IMO streaming files is a *really* common use-case that I wanted to show.